lib/engine/game/g_1877/step/buy_sell_par_shares.rb

```diff
@@ -113,12 +113,13 @@ def process_buy_train(action)
             track_action(action, entity, false)
             train = action.train
             price = action.price
+            source = train.owner
 
             raise GameError, 'Not a buyable train' unless buyable_train_variants(train,
                                                                                  entity).include?(train.variant)
             raise GameError, 'Must pay face value' if price != train.price
 
-            @game.queue_log! { @game.phase.buying_train!(entity, train) }
+            @game.queue_log! { @game.phase.buying_train!(entity, train, source) }
 
             source = @depot.discarded.include?(train) ? 'The Discard' : train.owner.name
```

Comment:
source_name

Reply:
and then used in the log message below.

Reply:
i used train.owner instead of source, no need to change this stuff now